ComfortZoneApp/app/src/main/java/com/example/comfortzone/flight/ui/CityMapViewFragment.java

```diff
@@ -58,11 +63,40 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         initViews(view);
+        setupSpinnerAdapter();
+        setSpinnerListener();
         createMap();
     }
 
     private void initViews(@NonNull View view) {
         mvMap = view.findViewById(R.id.mvMap);
+        spPinNum = view.findViewById(R.id.spPinNum);
+    }
+
+    public void setupSpinnerAdapter() {
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getContext(),
+                R.array.num_of_pins_array, android.R.layout.simple_spinner_item);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spPinNum.setAdapter(adapter);
+    }
+
+    public void setSpinnerListener() {
+        spPinNum.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                if (parent.getItemAtPosition(position).equals("All")) {
+                    maxCities = TOTAL_CITIES;
```

Comment:
is there a way to tell the map to render all without specifying the count so you don't need the total count?